src/dmd/dinifile.d

```diff
@@ -142,26 +142,21 @@ private bool writeToEnv(ref StringTable environment, char* nameEqValue)
  */
 void updateRealEnvironment(ref StringTable environment)
 {
-    static int envput(const(StringValue)* sv)
+    foreach (sv; environment)
     {
-        const name = sv.toDchars();
-        const namelen = strlen(name);
-        const value = cast(const(char)*)sv.ptrvalue;
+        const name = sv.toString();
+        const value = (cast(const(char)*)sv.ptrvalue)[0 .. sv.length];
         if (!value) // deleted?
```

Comment:
Should be `value.length` following the change

Reply:
Done, though it didn't resolve the segfaults.